client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js

```diff
@@ -376,6 +376,37 @@ angular.module('managerApp')
             self.states.hasOldFlavors = true;
           }
         });
+        // Remove flavors if OS has restricted
+        const restrictedFlavors = _.get(self.vmInEdition, 'image.flavorType', []);
+        const selectedFlavour = _.get(self.vmInEdition, 'flavor');
+        let selectedFlavourInFilterList = false;
+        if (!_.isEmpty(restrictedFlavors)) {
+          self.displayData.categories = _.filter(
+            self.displayData.categories,
+            (category) => {
+              const currentCategory = category;
+              currentCategory.flavors = _.filter(
+                category.flavors,
+                (flavor) => {
+                  const isRestricted = _.indexOf(restrictedFlavors, flavor.shortType) > -1;
+                  if (!selectedFlavourInFilterList
+                    && isRestricted
+                    && selectedFlavour.id === flavor.id) {
+                    selectedFlavourInFilterList = true;
```

Comment:
Please only use `filter` to filter, you'd better use `find` to find items

Reply:
I was not able to map your comment with code.  In this case, I am using _.filter to filter items from an array not to find an item. Are you sure your comment belongs to this code? can you please verify again.